### ISSUE TITLE
Powerdns record with count and hyphen in name fails on refresh

### ISSUE THREAD

**Issue by alkersan** [hashicorp/terraform#5228]
provider fails to retrieve record information if name contains hyphens.
```
resource "powerdns_record" "hyphened" {
	count = "2"
	zone = "example.org"
	name = "redis-${count.index}.example.org"
	type = "A"
	ttl = 60
	records = [ "10.20.30.${count.index}" ]
}
```
Reason: poor choice of `recordId` separator character [here](https://github.com/hashicorp/terraform/blob/master/builtin/providers/powerdns/client.go#L103).
Powerdns don't include unique identifier of rrsets, thus a tuple `(name, type)` was used to correlate terraform's resourceId with pdns record. But I totally missed the factor of allowed dns hostname characters and 've put allowed hyphen as separator between name and type, thus making impossible reverse parsing of resourceId like `redis-1.example.com-A`

While fix seems simple - to choose different separator (`*$#&`) that's not it [allowed set](https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names), i understand this is a complete FUBAR. Change of internal resource representation will make all existing records effectively invalid after update. Is there some internal migration mechanism in terraform that can be applied? But if not - I don't think this provider attracted lots of users, so impact shouldn't be huge.

**Comment by jen20** [comment]
Fixes #5228.

**Comment by jasonl** [comment]
For anyone who finds this, and needs to manually update their state file:

1. Open `terraform.tfstate` in a text editor
2. Find all the `powerdns_record` entries in it.
3. In each one, change **both** `id` keys from "your-domain.something.com-A" to "your-domain.something.com:::A".
4. Incremental the `serial` number at the top of `terraform.tfstate` so that Terraform doesn't think you've got a state file conflict.
5. Run `terraform plan`. It should show that no changes are required in your infrastructure.
